templates/job-dashboard.php

```diff
@@ -87,7 +87,7 @@
 				<?php endforeach; ?>
 			<?php endif; ?>
 		</tbody>
-		<?php if ( $submit_job_form_page_id && ( job_manager_count_user_job_listings() < $submission_limit || ! $submission_limit ) ) : ?>
+		<?php if ( $submit_job_form_page_id && ( job_manager_count_user_job_listings() < $submission_limit || 'false' === $submission_limit ) ) : ?>
 			<tfoot>
```

Comment:
There are other places in the plugin where this check needs to be updated. Might make sense to add a new `job_manager_can_user_submit_job()` to handle it consistently everywhere.

https://github.com/Automattic/WP-Job-Manager/blob/7411e91b491475e841a65c3b13b31132727c69f3/includes/class-wp-job-manager-shortcodes.php#L110-L114